galsim/interpolatedimage.py

```diff
@@ -683,13 +725,31 @@ def __init__(self, real_kimage, imag_kimage, k_interpolant=None, stepk=None,
             self._real_kimage.image, self._imag_kimage.image,
             self._real_kimage.scale, self._stepk, self.k_interpolant, gsparams))
 
+    def __eq__(self, other):
+        return (isinstance(other, galsim.InterpolatedKImage) and
+                self._real_kimage == other._real_kimage and
+                self._imag_kimage == other._imag_kimage and
+                self.k_interpolant == other.k_interpolant and
+                self._stepk == other._stepk and
+                self._gsparams == other._gsparams)
+
+    def __hash__(self):
+        # Definitely want to cache this, since the real and image kimages could be large.
+        if not hasattr(self, '_hash'):
+            self._hash = hash(("galsim.InterpolatedKImage", self.k_interpolant, self._stepk,
+                               self._gsparams))
+            for img in [self._real_kimage, self._imag_kimage]:
+                self._hash ^= hash(tuple(img.array.ravel()))
+                self._hash ^= hash((img.bounds, img.wcs))
+        return self._hash
+
     def __repr__(self):
-        return ('galsim.InterpolatedKImage(%r, %r, %r, stepk=%r, gsparams=%r)')%(
+        return ('galsim.InterpolatedKImage(\n%r,\n%r,\n%r, stepk=%r, gsparams=%r)') % (
             self._real_kimage, self._imag_kimage, self.k_interpolant,
             self._stepk, self._gsparams)
 
-    def __str__(self): return 'galsim.InterpolatedKImage(real_kimage=%s)'%(
-            self._real_kimage)
+    def __str__(self):
+        return 'galsim.InterpolatedKImage(real_kimage=%s)' % self._real_kimage
```

Comment:
Since imag_kimage has to have the same metadata has real_kimage, I don't think we lost much without it, but I guess it's good to remind users that both are required for InterpolatedKImage.  Fixed.